src/gui/sdl_gui.cpp

```diff
@@ -1784,6 +1786,60 @@ void GFX_Init()
 	        SDL_GetCurrentVideoDriver(),
 	        SDL_GetCurrentAudioDriver());
 
+#ifdef MACOSX
+	// On macOS, check for document drop events (double-click to open .dosbox packages)
+	// When a user double-clicks a .dosbox package in Finder, macOS delivers the
+	// file path via SDL_DROPFILE event rather than command-line arguments.
+	// We check for this immediately after SDL_Init but before the restart function
+	// is registered, so we implement the restart directly here.
+	// Sleep briefly to allow the OS time to queue the drop event before we poll.
+	SDL_Delay(100);
+	SDL_Event event;
+	while (SDL_PollEvent(&event)) {
+		if (event.type == SDL_DROPFILE && event.drop.file != nullptr) {
+			const std::string dropped_file_path = event.drop.file;
+			SDL_free(event.drop.file);
+
+			LOG_MSG("CONFIG: Received dropped file via SDL_DROPFILE: '%s'",
+			        dropped_file_path.c_str());
+
+			// Check if it's a .dosbox package
+			if (is_macos_dosbox_package(dropped_file_path)) {
+				LOG_MSG("CONFIG: Detected .dosbox package, scanning for configuration files");
+
+				// Scan the package for .conf files and get absolute paths
+				auto package = scan_macos_dosbox_package(dropped_file_path);
+
+				if (!package.is_valid) {
+					LOG_WARNING("CONFIG: Failed to scan .dosbox package");
+					continue;
+				}
+
+				LOG_MSG("CONFIG: Found %zu configuration file(s)", package.conf_files.size());
+
+				// Build new startup parameters with expanded package
+				auto new_params = control->startup_params;
+
+				// Add --working-dir to set the game directory
+				new_params.push_back("--working-dir");
+				new_params.push_back(package.package_path);
+
+				// Add --conf for each configuration file found
+				for (const auto& conf_file : package.conf_files) {
+					new_params.push_back("--conf");
+					new_params.push_back(conf_file);
+				}
```

Comment:
Okay, so let's simplify things: 

- We're only checking for `dosbox.conf` at the root level in the package now.
- So we don't need to provide the `--conf` settings, we just need to set `--working-dir` and that's it. `dosbox.conf` will be then picked up automatically, no need for the `--conf` arg.
- We only need `is_macos_dosbox_package()` now because things are a lot simpler (`scan_macos_dosbox_package()` can be merged into `is_macos_dosbox_package()`).
- The `dosbox_macos_package` will fall away too; we just need to pass the `std_fs::path` to the package around.